[32mINFO    [0m juju.model:model.py:2098 Deploying ch:amd64/jammy/self-signed-certificates-155
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] active: 
  shard-one/1 [idle] active: 
  shard-one/2 [idle] active: 
  shard-two/0 [idle] active: Primary
  config-server/0 [idle] active: 
  config-server/1 [idle] active: Primary
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] active: 
  shard-one/1 [idle] active: 
  shard-one/2 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] blocked: Shard requires TLS to be enabled.
  shard-one/1 [idle] blocked: Shard requires TLS to be enabled.
  shard-one/2 [idle] blocked: Shard requires TLS to be enabled.
  shard-two/0 [idle] active: Primary
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] blocked: Shard requires TLS to be enabled.
  shard-one/1 [idle] blocked: Shard requires TLS to be enabled.
  shard-one/2 [idle] blocked: Shard requires TLS to be enabled.
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] blocked: shards shard-one are unreachable.
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] blocked: Shard requires TLS to be enabled.
  shard-one/1 [idle] blocked: Shard requires TLS to be enabled.
  shard-one/2 [idle] blocked: Shard requires TLS to be enabled.
  config-server/0 [executing] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [executing] active: 
  shard-one/1 [executing] maintenance: enabling TLS
  shard-one/2 [idle] active: 
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] blocked: shards shard-one are unreachable.
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/0 [idle] active: 
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] active: 
  shard-one/1 [idle] active: 
  shard-one/2 [idle] active: 
  shard-two/0 [idle] active: Primary
  config-server/0 [executing] active: 
  config-server/1 [executing] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-one/1 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-one/2 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-two/0 [idle] active: Primary
  config-server/0 [idle] active: 
  config-server/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-one/1 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-one/2 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-two/0 [idle] active: Primary
  config-server/0 [executing] active: 
  config-server/1 [executing] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  shard-one/0 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-one/1 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-one/2 [idle] blocked: Shard has TLS enabled, but config-server does not.
  shard-two/0 [idle] active: Primary
  config-server/0 [executing] active: 
  config-server/1 [executing] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  config-server/1 [idle] active:
[32mINFO    [0m pytest_operator.plugin:plugin.py:834 Model status:

Model  Controller          Cloud/Region        Version  SLA          Timestamp
test   microk8s-localhost  microk8s/localhost  3.5.3    unsupported  02:15:11Z

App                                Version  Status   Scale  Charm                     Channel        Rev  Address         Exposed  Message
config-server                               active       2  mongodb-k8s                                3  10.152.183.244  no       
self-signed-certificates                    active       1  self-signed-certificates  latest/stable  155  10.152.183.64   no       
self-signed-certificates-separate           active       1  self-signed-certificates  latest/stable  155  10.152.183.206  no       
shard-one                                   blocked      3  mongodb-k8s                                4  10.152.183.211  no       Shard CA and Config-Server CA don't match.
shard-two                                   blocked      1  mongodb-k8s                                5  10.152.183.154  no       Shard CA and Config-Server CA don't match.

Unit                                  Workload  Agent  Address      Ports  Message
config-server/0                       active    idle   10.1.41.225         
config-server/1*                      active    idle   10.1.41.224         
self-signed-certificates-separate/0*  active    idle   10.1.41.238         
self-signed-certificates/0*           active    idle   10.1.41.218         
shard-one/0*                          blocked   idle   10.1.41.232         Shard CA and Config-Server CA don't match.
shard-one/1                           blocked   idle   10.1.41.236         Shard CA and Config-Server CA don't match.
shard-one/2                           blocked   idle   10.1.41.235         Shard CA and Config-Server CA don't match.
shard-two/0*                          blocked   idle   10.1.41.237         Shard CA and Config-Server CA don't match.

[32mINFO    [0m pytest_operator.plugin:plugin.py:840 Juju error logs:

unit-config-server-0: 01:38:57 ERROR unit.config-server/0.juju-log database-peers:0: Failed to create the operator user: non-zero exit code 1 executing ['/usr/bin/mongosh', 'mongodb://localhost/admin', '--quiet', '--eval', "db.createUser({  user: 'operator',  pwd: passwordPrompt(),  roles:[    {'role': 'userAdminAnyDatabase', 'db': 'admin'},     {'role': 'readWriteAnyDatabase', 'db': 'admin'},     {'role': 'clusterAdmin', 'db': 'admin'},   ],  mechanisms: ['SCRAM-SHA-256'],  passwordDigestor: 'server',})"], stdout='Enter password\n********************************', stderr='MongoServerError: not primary\n'
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-config-server-0/charm/./src/charm.py", line 939, in _init_operator_user
    process.wait_output()
  File "/var/lib/juju/agents/unit-config-server-0/charm/venv/ops/pebble.py", line 1773, in wait_output
    raise ExecError[AnyStr](self._command, exit_code, out_value, err_value)
ops.pebble.ExecError: non-zero exit code 1 executing ['/usr/bin/mongosh', 'mongodb://localhost/admin', '--quiet', '--eval', "db.createUser({  user: 'operator',  pwd: passwordPrompt(),  roles:[    {'role': 'userAdminAnyDatabase', 'db': 'admin'},     {'role': 'readWriteAnyDatabase', 'db': 'admin'},     {'role': 'clusterAdmin', 'db': 'admin'},   ],  mechanisms: ['SCRAM-SHA-256'],  passwordDigestor: 'server',})"], stdout='Enter password\n********************************', stderr='MongoServerError: not primary\n'
unit-config-server-0: 01:39:03 ERROR unit.config-server/0.juju-log database-peers:0: Failed to create the operator user: non-zero exit code 1 executing ['/usr/bin/mongosh', 'mongodb://localhost/admin', '--quiet', '--eval', "db.createUser({  user: 'operator',  pwd: passwordPrompt(),  roles:[    {'role': 'userAdminAnyDatabase', 'db': 'admin'},     {'role': 'readWriteAnyDatabase', 'db': 'admin'},     {'role': 'clusterAdmin', 'db': 'admin'},   ],  mechanisms: ['SCRAM-SHA-256'],  passwordDigestor: 'server',})"], stdout='Enter password\n********************************', stderr='MongoServerError: not primary\n'
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-config-server-0/charm/./src/charm.py", line 939, in _init_operator_user
    process.wait_output()
  File "/var/lib/juju/agents/unit-config-server-0/charm/venv/ops/pebble.py", line 1773, in wait_output
    raise ExecError[AnyStr](self._command, exit_code, out_value, err_value)
ops.pebble.ExecError: non-zero exit code 1 executing ['/usr/bin/mongosh', 'mongodb://localhost/admin', '--quiet', '--eval', "db.createUser({  user: 'operator',  pwd: passwordPrompt(),  roles:[    {'role': 'userAdminAnyDatabase', 'db': 'admin'},     {'role': 'readWriteAnyDatabase', 'db': 'admin'},     {'role': 'clusterAdmin', 'db': 'admin'},   ],  mechanisms: ['SCRAM-SHA-256'],  passwordDigestor: 'server',})"], stdout='Enter password\n********************************', stderr='MongoServerError: not primary\n'
unit-config-server-0: 01:40:35 ERROR unit.config-server/0.juju-log config-server:3: Failed to add shard shard-one to the config server, error=OperationFailure("Authentication failed., full error: {'ok': 0.0, 'errmsg': 'Authentication failed.', 'code': 18, 'codeName': 'AuthenticationFailed', '$clusterTime': {'clusterTime': Timestamp(1725673234, 2), 'signature': {'hash': b'\\xb3!\\x14B\\x9e\\x87\\x11\\r;\\xa9\\x97\\xb2\\xa3\\xb6+\\x9al\\xb7\\x93\\xb4', 'keyId': 7411709734245367832}}, 'operationTime': Timestamp(1725673234, 2)}")
unit-config-server-0: 01:40:35 ERROR unit.config-server/0.juju-log config-server:3: shard-one shard does not have the same auth as the config server.
unit-config-server-0: 01:40:36 ERROR unit.config-server/0.juju-log config-server:3: Failed to add shard shard-one to the config server, error=OperationFailure("Authentication failed., full error: {'ok': 0.0, 'errmsg': 'Authentication failed.', 'code': 18, 'codeName': 'AuthenticationFailed', '$clusterTime': {'clusterTime': Timestamp(1725673235, 1), 'signature': {'hash': b'4hAd\\x88v\\x15\\xdeA\\xcdt\\xf9Z\\xcaE\\x1f~.\\x1c\\x8e', 'keyId': 7411709734245367832}}, 'operationTime': Timestamp(1725673235, 1)}")
unit-config-server-0: 01:40:36 ERROR unit.config-server/0.juju-log config-server:3: shard-one shard does not have the same auth as the config server.
unit-config-server-0: 01:41:00 ERROR unit.config-server/0.juju-log config-server:4: Failed to add shard shard-two to the config server, error=OperationFailure("Authentication failed., full error: {'ok': 0.0, 'errmsg': 'Authentication failed.', 'code': 18, 'codeName': 'AuthenticationFailed', '$clusterTime': {'clusterTime': Timestamp(1725673260, 1), 'signature': {'hash': b'\\x137\\x1e\\xba\\x98\\xca\\x05\\xe6\\x18\\x99\\x1e\\xa5\\x12\\x9d3a\\xac\\xcf(\\x19', 'keyId': 7411709734245367832}}, 'operationTime': Timestamp(1725673260, 1)}")
unit-config-server-0: 01:41:00 ERROR unit.config-server/0.juju-log config-server:4: shard-two shard does not have the same auth as the config server.
unit-shard-one-2: 01:45:27 ERROR unit.shard-one/2.juju-log certificates:5: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-1: 01:45:27 ERROR unit.shard-one/1.juju-log certificates:5: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-0: 01:45:27 ERROR unit.shard-one/0.juju-log certificates:5: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-two-0: 01:45:27 ERROR unit.shard-two/0.juju-log certificates:6: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-2: 01:45:28 ERROR unit.shard-one/2.juju-log certificates:5: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-0: 01:45:28 ERROR unit.shard-one/0.juju-log certificates:5: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-1: 01:45:28 ERROR unit.shard-one/1.juju-log certificates:5: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-1: 01:45:28 ERROR unit.config-server/1.juju-log certificates:7: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-two-0: 01:45:28 ERROR unit.shard-two/0.juju-log certificates:6: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-0: 01:45:28 ERROR unit.config-server/0.juju-log certificates:7: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-config-server-1: 01:45:29 ERROR unit.config-server/1.juju-log certificates:7: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-0: 01:45:29 ERROR unit.config-server/0.juju-log certificates:7: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-1: 01:45:42 ERROR unit.config-server/1.juju-log certificates:7: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 01:45:42 ERROR unit.config-server/1.juju-log certificates:7: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 01:45:57 ERROR unit.config-server/1.juju-log This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 01:45:57 ERROR unit.config-server/1.juju-log This operation (update_relation_data()) can only be performed by the leader unit
unit-shard-one-0: 01:46:28 ERROR unit.shard-one/0.juju-log Unable to access primary due to: No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 66dbb0735442dcdbe096913f, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-0.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.00048599900003409857>, <ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0007722510000576222>, <ServerDescription ('shard-one-2.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0009832970000616115>]>
unit-shard-one-0: 01:46:31 ERROR unit.shard-one/0.juju-log Unable to access primary due to: No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 66dbb076302f773cfe0edfc8, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-0.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.004330358999936834>, <ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=NotPrimaryError('The server is in quiesce mode and will shut down, full error: {\'topologyVersion\': {\'processId\': ObjectId(\'66dbb071d145b5caedf442d0\'), \'counter\': 4}, \'ok\': 0.0, \'errmsg\': \'The server is in quiesce mode and will shut down\', \'code\': 91, \'codeName\': \'ShutdownInProgress\', \'remainingQuiesceTimeMillis\': 14993, \'lastCommittedOpTime\': Timestamp(0, 0), \'$clusterTime\': {\'clusterTime\': Timestamp(1725673586, 9), \'signature\': {\'hash\': b\'r\\xf3S\\xd7\\xdf\\xdf\\x89\\x94t"a!?\\xc9\\x88\\xfdI16\\xf5\', \'keyId\': 7411709734245367832}}, \'operationTime\': Timestamp(1725673532, 1)}')>, <ServerDescription ('shard-one-2.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.0019104959999367566>]>
unit-shard-one-0: 01:46:34 ERROR unit.shard-one/0.juju-log Unable to access primary due to: No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 66dbb0794dc32852cb56bf36, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-0.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.00039451899999676243>, <ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=NotPrimaryError('The server is in quiesce mode and will shut down, full error: {\'topologyVersion\': {\'processId\': ObjectId(\'66dbb071d145b5caedf442d0\'), \'counter\': 4}, \'ok\': 0.0, \'errmsg\': \'The server is in quiesce mode and will shut down\', \'code\': 91, \'codeName\': \'ShutdownInProgress\', \'remainingQuiesceTimeMillis\': 12434, \'lastCommittedOpTime\': Timestamp(0, 0), \'$clusterTime\': {\'clusterTime\': Timestamp(1725673586, 9), \'signature\': {\'hash\': b\'r\\xf3S\\xd7\\xdf\\xdf\\x89\\x94t"a!?\\xc9\\x88\\xfdI16\\xf5\', \'keyId\': 7411709734245367832}}, \'operationTime\': Timestamp(1725673532, 1)}')>, <ServerDescription ('shard-one-2.shard-one-endpoints', 27017) server_type: RSSecondary, rtt: 0.00039838500003952504>]>
unit-shard-one-0: 01:47:44 ERROR unit.shard-one/0.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-0: 01:47:44 ERROR unit.shard-one/0.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-1: 01:47:46 ERROR unit.shard-one/1.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-1: 01:47:46 ERROR unit.shard-one/1.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-2: 01:48:04 ERROR unit.shard-one/2.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-2: 01:48:05 ERROR unit.shard-one/2.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-0: 01:48:05 ERROR unit.shard-one/0.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-0: 01:48:05 ERROR unit.shard-one/0.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-1: 01:48:07 ERROR unit.shard-one/1.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-one-1: 01:48:07 ERROR unit.shard-one/1.juju-log certificates:5: An unknown certificate is available -- ignoring.
unit-shard-two-0: 01:49:55 ERROR unit.shard-two/0.juju-log certificates:6: An unknown certificate is available -- ignoring.
unit-shard-two-0: 01:49:55 ERROR unit.shard-two/0.juju-log certificates:6: An unknown certificate is available -- ignoring.
unit-config-server-0: 01:51:34 ERROR unit.config-server/0.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-0: 01:51:34 ERROR unit.config-server/0.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:51:36 ERROR unit.config-server/1.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:51:36 ERROR unit.config-server/1.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:51:37 ERROR unit.config-server/1.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:51:37 ERROR unit.config-server/1.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:51:37 ERROR unit.config-server/1.juju-log certificates:7: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-1: 01:51:37 ERROR unit.config-server/1.juju-log certificates:7: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-0: 01:51:54 ERROR unit.config-server/0.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-0: 01:51:54 ERROR unit.config-server/0.juju-log certificates:7: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:53:41 ERROR unit.config-server/1.juju-log certificates:7: This operation (delete_relation_data()) can only be performed by the leader unit
unit-config-server-1: 01:53:41 ERROR unit.config-server/1.juju-log certificates:7: This operation (delete_relation_data()) can only be performed by the leader unit
unit-self-signed-certificates-0: 01:53:55 ERROR juju.worker.dependency "uniter" manifold worker returned unexpected error: relation scope id "2e28f05b-8481-42b6-8689-6cf5d95549ea:r#7#config-server": settings 2e28f05b-8481-42b6-8689-6cf5d95549ea:r#7#config-server not found (not found)
unit-config-server-1: 01:53:56 ERROR juju.worker.dependency "uniter" manifold worker returned unexpected error: relation scope id "2e28f05b-8481-42b6-8689-6cf5d95549ea:r#7#self-signed-certificates": settings 2e28f05b-8481-42b6-8689-6cf5d95549ea:r#7#self-signed-certificates not found (not found)
unit-config-server-0: 01:53:57 ERROR juju.worker.dependency "uniter" manifold worker returned unexpected error: relation scope id "2e28f05b-8481-42b6-8689-6cf5d95549ea:r#7#self-signed-certificates": settings 2e28f05b-8481-42b6-8689-6cf5d95549ea:r#7#self-signed-certificates not found (not found)
unit-shard-two-0: 01:53:58 ERROR unit.shard-two/0.juju-log Unable to access primary due to: shard-two-0.shard-two-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66dbb235913e841aa7b4c958, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-two-0.shard-two-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-two-0.shard-two-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>
unit-shard-one-0: 01:53:58 ERROR unit.shard-one/0.juju-log Unable to access primary due to: shard-one-1.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),shard-one-0.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),shard-one-2.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66dbb235cf834e72f826ff62, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-0.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-one-0.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-one-1.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('shard-one-2.shard-one-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-one-2.shard-one-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>
unit-shard-two-0: 01:54:00 ERROR unit.shard-two/0.juju-log Unable to access primary due to: shard-two-0.shard-two-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66dbb23704c387d7d806f85e, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-two-0.shard-two-endpoints', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('shard-two-0.shard-two-endpoints:27017: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>
unit-config-server-0: 01:54:03 ERROR unit.config-server/0.juju-log config-server:3: Deferring _on_relation_event for shards interface since: error=ServerSelectionTimeoutError("config-server-1.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),config-server-0.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66dbb23a5bd7a9f3fb30b76c, topology_type: Unknown, servers: [<ServerDescription ('config-server-0.config-server-endpoints', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('config-server-0.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('config-server-1.config-server-endpoints', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('config-server-1.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>")
unit-config-server-0: 01:54:06 ERROR unit.config-server/0.juju-log Deferring _on_relation_event for shards interface since: error=ServerSelectionTimeoutError("config-server-1.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),config-server-0.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66dbb23d862828e94f43ebee, topology_type: Unknown, servers: [<ServerDescription ('config-server-0.config-server-endpoints', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('config-server-0.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('config-server-1.config-server-endpoints', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('config-server-1.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>")
unit-shard-one-1: 01:54:29 ERROR unit.shard-one/1.juju-log shard-one/1.is_unit_in_replica_set PyMongoError=No replica set members match selector "Primary()", Timeout: 1.0s, Topology Description: <TopologyDescription id: 66dbb2530b7c589e4e3807cd, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('shard-one-1.shard-one-endpoints', 27017) server_type: RSGhost, rtt: 0.009269590000030803>]>
controller-0: 01:55:48 ERROR juju.worker.caasapplicationprovisioner.runner exited "shard-one": finding filesystem info for shard-one-mongodb-logs-23b1adef: unable to get persistent volume claim: k8s: persistentvolumeclaims "shard-one-mongodb-logs-23b1adef-shard-one-1" not found
unit-shard-one-1: 01:57:24 ERROR unit.shard-one/1.juju-log certificates:11: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-0: 01:57:25 ERROR unit.shard-one/0.juju-log certificates:11: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-config-server-0: 01:57:25 ERROR unit.config-server/0.juju-log certificates:13: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-2: 01:57:25 ERROR unit.shard-one/2.juju-log certificates:11: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-config-server-1: 01:57:25 ERROR unit.config-server/1.juju-log certificates:13: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-1: 01:57:25 ERROR unit.shard-one/1.juju-log certificates:11: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-0: 01:57:25 ERROR unit.config-server/0.juju-log certificates:13: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-two-0: 01:57:25 ERROR unit.shard-two/0.juju-log certificates:12: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-0: 01:57:25 ERROR unit.shard-one/0.juju-log certificates:11: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-1: 01:57:26 ERROR unit.config-server/1.juju-log certificates:13: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-two-0: 01:57:26 ERROR unit.shard-two/0.juju-log certificates:12: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-2: 01:57:26 ERROR unit.shard-one/2.juju-log certificates:11: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-1: 01:58:53 ERROR unit.config-server/1.juju-log config-server:14: Failed to add shard shard-one to the config server, error=OperationFailure('Could not find user "O=config-server,x500UniqueIdentifier=79213553-2def-4fce-b17f-396a823ad4aa,CN=config-server" for db "$external", full error: {\'ok\': 0.0, \'errmsg\': \'Could not find user "O=config-server,x500UniqueIdentifier=79213553-2def-4fce-b17f-396a823ad4aa,CN=config-server" for db "$external"\', \'code\': 11, \'codeName\': \'UserNotFound\', \'$clusterTime\': {\'clusterTime\': Timestamp(1725674333, 1), \'signature\': {\'hash\': b\'\\xa2\\x86\\xb7"f\\xd4\\x10\\xb8xt\\x80\\x85\\xf8\\x12\\x9fC \\xd8i\\xcd\', \'keyId\': 7411714201011355674}}, \'operationTime\': Timestamp(1725674332, 1)}')
unit-config-server-1: 01:58:54 ERROR unit.config-server/1.juju-log config-server:14: Failed to add shard-one to cluster
unit-config-server-1: 01:58:54 ERROR unit.config-server/1.juju-log config-server:14: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find user "O=config-server,x500UniqueIdentifier=79213553-2def-4fce-b17f-396a823ad4aa,CN=config-server" for db "$external", full error: {\'ok\': 0.0, \'errmsg\': \'Could not find user "O=config-server,x500UniqueIdentifier=79213553-2def-4fce-b17f-396a823ad4aa,CN=config-server" for db "$external"\', \'code\': 11, \'codeName\': \'UserNotFound\', \'$clusterTime\': {\'clusterTime\': Timestamp(1725674333, 1), \'signature\': {\'hash\': b\'\\xa2\\x86\\xb7"f\\xd4\\x10\\xb8xt\\x80\\x85\\xf8\\x12\\x9fC \\xd8i\\xcd\', \'keyId\': 7411714201011355674}}, \'operationTime\': Timestamp(1725674332, 1)}')
unit-config-server-1: 01:58:55 ERROR unit.config-server/1.juju-log config-server:15: Failed to add shard shard-two to the config server, error=OperationFailure('Could not find user "O=config-server,x500UniqueIdentifier=79213553-2def-4fce-b17f-396a823ad4aa,CN=config-server" for db "$external", full error: {\'ok\': 0.0, \'errmsg\': \'Could not find user "O=config-server,x500UniqueIdentifier=79213553-2def-4fce-b17f-396a823ad4aa,CN=config-server" for db "$external"\', \'code\': 11, \'codeName\': \'UserNotFound\', \'$clusterTime\': {\'clusterTime\': Timestamp(1725674334, 2), \'signature\': {\'hash\': b\'\\xddq\\x1c\\xcf\\xc5n\\xeb\\xaeQE\\x06\\x97\\x1cdG\\x17]9Jq\', \'keyId\': 7411714201011355674}}, \'operationTime\': Timestamp(1725674334, 2)}')
unit-config-server-1: 01:59:10 ERROR unit.config-server/1.juju-log config-server:15: Failed to add shard shard-one to the config server, error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1725674349, 2), \'signature\': {\'hash\': b\'\\x18\\xca\\xe8Ne\\xe1\\x9b\\x8es\\x85\\xfb\\xe0F\\x01K\\r#\\xd8\\x92\\xd6\', \'keyId\': 7411714201011355674}}, \'operationTime\': Timestamp(1725674349, 2)}')
unit-config-server-1: 01:59:10 ERROR unit.config-server/1.juju-log config-server:15: Failed to add shard-one to cluster
unit-config-server-1: 01:59:10 ERROR unit.config-server/1.juju-log config-server:15: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1725674349, 2), \'signature\': {\'hash\': b\'\\x18\\xca\\xe8Ne\\xe1\\x9b\\x8es\\x85\\xfb\\xe0F\\x01K\\r#\\xd8\\x92\\xd6\', \'keyId\': 7411714201011355674}}, \'operationTime\': Timestamp(1725674349, 2)}')
unit-config-server-1: 01:59:12 ERROR unit.config-server/1.juju-log config-server:14: Failed to add shard shard-two to the config server, error=OperationFailure('Could not find user "O=config-server,x500UniqueIdentifier=79213553-2def-4fce-b17f-396a823ad4aa,CN=config-server" for db "$external", full error: {\'ok\': 0.0, \'errmsg\': \'Could not find user "O=config-server,x500UniqueIdentifier=79213553-2def-4fce-b17f-396a823ad4aa,CN=config-server" for db "$external"\', \'code\': 11, \'codeName\': \'UserNotFound\', \'$clusterTime\': {\'clusterTime\': Timestamp(1725674351, 1), \'signature\': {\'hash\': b\'\\x89Y\\xac\\xda\\xdd\\x0e D\\xe6Z4O\\xf6\\xc6\\x8e\\x91I)\\xb1\\xeb\', \'keyId\': 7411714201011355674}}, \'operationTime\': Timestamp(1725674351, 1)}')
unit-shard-one-1: 01:59:12 ERROR unit.shard-one/1.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-one-2: 01:59:12 ERROR unit.shard-one/2.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-one-1: 01:59:12 ERROR unit.shard-one/1.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-one-2: 01:59:12 ERROR unit.shard-one/2.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-one-0: 01:59:15 ERROR unit.shard-one/0.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-one-0: 01:59:15 ERROR unit.shard-one/0.juju-log certificates:11: An unknown certificate is available -- ignoring.
unit-shard-two-0: 01:59:24 ERROR unit.shard-two/0.juju-log certificates:12: An unknown certificate is available -- ignoring.
unit-shard-two-0: 01:59:24 ERROR unit.shard-two/0.juju-log certificates:12: An unknown certificate is available -- ignoring.
unit-config-server-1: 01:59:27 ERROR unit.config-server/1.juju-log config-server:14: Failed to add shard shard-one to the config server, error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1725674366, 1), \'signature\': {\'hash\': b\'\\x9e1\\xe3\\x9ff3(W\\xbb\\xccf\\xec\\x9c\\x10\\xa3\\x04/\\xb6\\xb7\\xda\', \'keyId\': 7411714201011355674}}, \'operationTime\': Timestamp(1725674366, 1)}')
unit-config-server-1: 01:59:27 ERROR unit.config-server/1.juju-log config-server:14: Failed to add shard-one to cluster
unit-config-server-1: 01:59:27 ERROR unit.config-server/1.juju-log config-server:14: Deferring _on_relation_event for shards interface since: error=OperationFailure('Could not find host matching read preference { mode: "primary" } for set shard-one, full error: {\'ok\': 0.0, \'errmsg\': \'Could not find host matching read preference { mode: "primary" } for set shard-one\', \'code\': 133, \'codeName\': \'FailedToSatisfyReadPreference\', \'$clusterTime\': {\'clusterTime\': Timestamp(1725674366, 1), \'signature\': {\'hash\': b\'\\x9e1\\xe3\\x9ff3(W\\xbb\\xccf\\xec\\x9c\\x10\\xa3\\x04/\\xb6\\xb7\\xda\', \'keyId\': 7411714201011355674}}, \'operationTime\': Timestamp(1725674366, 1)}')
unit-shard-one-0: 02:08:21 ERROR unit.shard-one/0.juju-log certificates:16: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-0: 02:08:21 ERROR unit.shard-one/0.juju-log certificates:16: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-1: 02:08:24 ERROR unit.shard-one/1.juju-log certificates:16: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-1: 02:08:24 ERROR unit.shard-one/1.juju-log certificates:16: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-shard-one-2: 02:08:25 ERROR unit.shard-one/2.juju-log certificates:16: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-shard-one-2: 02:08:26 ERROR unit.shard-one/2.juju-log certificates:16: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-0: 02:11:56 ERROR unit.config-server/0.juju-log certificates:13: This operation (delete_relation_data()) can only be performed by the leader unit
unit-config-server-0: 02:11:56 ERROR unit.config-server/0.juju-log certificates:13: This operation (delete_relation_data()) can only be performed by the leader unit
unit-config-server-1: 02:12:20 ERROR unit.config-server/1.juju-log config-server:14: Deferring _on_relation_event for shards interface since: error=ServerSelectionTimeoutError("config-server-1.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms),config-server-0.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms), Timeout: 1.0s, Topology Description: <TopologyDescription id: 66dbb683e9133b22ef1d7bfa, topology_type: Unknown, servers: [<ServerDescription ('config-server-0.config-server-endpoints', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('config-server-0.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>, <ServerDescription ('config-server-1.config-server-endpoints', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('config-server-1.config-server-endpoints:27018: [Errno 111] Connection refused (configured timeouts: socketTimeoutMS: 2000.0ms, connectTimeoutMS: 2000.0ms)')>]>")
unit-config-server-0: 02:12:47 ERROR unit.config-server/0.juju-log certificates:17: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-config-server-1: 02:12:47 ERROR unit.config-server/1.juju-log certificates:17: Non-existing secret unit:int-cert-secret was attempted to be removed.
unit-config-server-1: 02:12:48 ERROR unit.config-server/1.juju-log certificates:17: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-0: 02:12:48 ERROR unit.config-server/0.juju-log certificates:17: Non-existing secret unit:ext-cert-secret was attempted to be removed.
unit-config-server-0: 02:12:52 ERROR unit.config-server/0.juju-log certificates:17: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-0: 02:12:52 ERROR unit.config-server/0.juju-log certificates:17: This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-0: 02:13:08 ERROR unit.config-server/0.juju-log This operation (update_relation_data()) can only be performed by the leader unit
unit-config-server-0: 02:13:08 ERROR unit.config-server/0.juju-log This operation (update_relation_data()) can only be performed by the leader unit
unit-shard-two-0: 02:15:02 ERROR unit.shard-two/0.juju-log Shard is integrated to a different CA than the config server. Please use the same CA for all cluster components.
unit-shard-one-2: 02:15:07 ERROR unit.shard-one/2.juju-log Shard is integrated to a different CA than the config server. Please use the same CA for all cluster components.
unit-shard-one-1: 02:15:07 ERROR unit.shard-one/1.juju-log Shard is integrated to a different CA than the config server. Please use the same CA for all cluster components.
unit-shard-one-0: 02:15:08 ERROR unit.shard-one/0.juju-log Shard is integrated to a different CA than the config server. Please use the same CA for all cluster components.

[32mINFO    [0m pytest_operator.plugin:plugin.py:904 Forgetting main...